mne/io/eeglab/tests/test_eeglab.py

```diff
@@ -105,6 +107,27 @@ def test_io_set_raw(fname):
         assert_array_equal(raw0.annotations.duration, 0.0)
 
 
+@testing.requires_testing_data
+def test_io_set_preload_false_uses_lazy_loading():
+    """Ensure reading .set files without preload keeps data out of memory."""
+    raw_preloaded = read_raw_eeglab(raw_fname_mat, preload=True)
+    raw_not_preloaded = read_raw_eeglab(raw_fname_mat, preload=False)
+
+    assert not raw_not_preloaded.preload
+    assert getattr(raw_not_preloaded, "_data", None) is None
+    assert raw_not_preloaded.n_times == raw_preloaded.n_times
+    assert raw_not_preloaded.info["nchan"] == raw_preloaded.info["nchan"]
+
+    lazy_slice = raw_not_preloaded[:2, :10][0]
+    assert lazy_slice.shape == (2, 10)
+
+    # on-demand reads must not flip the preload flag or populate _data
+    assert not raw_not_preloaded.preload
+    assert getattr(raw_not_preloaded, "_data", None) is None
+
+    assert raw_preloaded._size > raw_not_preloaded._size
```

Comment:
Somewhere else in this file there is a `_test_raw_reader(..., test_preloading=False)` that can now become a `_test_raw_reader(...)` (the default is `test_preloading=True`)

https://github.com/mne-tools/mne-python/blob/3cfac64bbc778b0c52dcd76f4b4799cc59226fae/mne/io/eeglab/tests/test_eeglab.py#L644